### DISCUSSION TITLE
Vue Slot Props?

### DISCUSSION THREAD

**Discussion by jiblett1000** [chakra-ui/ark#788] [Q&A] [answered]
Hey there,

Just wondering if slot props to expose state will become available? I know development is still pretty early on. Just curious. For example:

```html
  <Select
    v-model="selectedOption"
    v-slot="{ isOpen }"
  >
    <SelectLabel>Label</SelectLabel>
    <SelectTrigger>
      <button>
        {{ selectedOption?.label ?? "Select" }}
        <ArrowDropDownIcon
          :class="[
            'h-6 w-6 flex-none transition-transform duration-150',
            isOpen && 'rotate-180',
          ]"
        />
      </button>
    </SelectTrigger>
    
    ...
 </Select>
```

**Comment by Shyrro** [comment, marked as answer]
Hey @jiblett1000 ! 

Actually yes. In other components we already expose the state, either through template refs or through slot props. 
Depends on the component, we missed the select so we'll definetly update this one 🙏

As for the rest, we'll always try to expose relevant state as much possible, either through slot props or template refs!

Hope that answers your question ☺️

**Reply by jiblett1000** [reply]
@Shyrro Awesome. That does answer it. I played around with zag-js at first but am now switching things over to Ark. Guess I haven't come across any slot props with the 1 or 2 I've done so far. Anyhow, great to hear. Thanks!
